src/fidesops/api/v1/endpoints/connection_type_endpoints.py

```diff
@@ -45,4 +46,43 @@ def is_match(elem: str) -> bool:
     ]
     connection_types: List[str] = sorted(database_types + saas_types)
 
+    return connection_types
+
+
+@router.get(
+    CONNECTION_TYPES,
+    dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_TYPE_READ])],
+    response_model=Page[str],
+)
+def get_all_connection_types(
+    *, params: Params = Depends(), search: Optional[str] = None
+) -> AbstractPage[str]:
+    """Returns a list of connection options in Fidesops - includes only database and saas options here."""
+
+    connection_types: List[str] = get_connection_types(search)
+
     return paginate(connection_types, params)
+
+
+@router.get(
+    CONNECTION_TYPE_SECRETS,
+    dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_TYPE_READ])],
+)
+def get_connection_type_secret_schema(
+    *, connection_type: str
+) -> Optional[Dict[str, Any]]:
+    """Returns the secret fields that should be supplied to authenticate with a particular connection type"""
```

Comment:
This looks good. It might be worth pointing out in a comment somewhere that we should never be loading any specific DB object into the schemas returned by this endpoint.